inc/classes/Addons/Cloudflare/Cloudflare.php

```diff
@@ -0,0 +1,636 @@
+
+			if ( empty( $cf_settings->success ) ) {
+				foreach ( $cf_settings->errors as $error ) {
+					$errors[] = $error->message;
+				}
+
+				$errors = wp_sprintf_l( '%l ', $errors );
+				return new \WP_Error( 'cloudflare_dev_mode', $errors );
+			}
+
+			foreach ( $cf_settings->result as $cloudflare_option ) {
+				switch ( $cloudflare_option->id ) {
+					case 'browser_cache_ttl':
+						$browser_cache_ttl = $cloudflare_option->value;
+						break;
+					case 'cache_level':
+						$cache_level = $cloudflare_option->value;
+						break;
+					case 'rocket_loader':
+						$rocket_loader = $cloudflare_option->value;
+						break;
+					case 'minify':
+						$cf_minify = $cloudflare_option->value;
+						break;
+				}
+			}
+			$cf_minify_value = 'on';
+
+			if ( 'off' === $cf_minify->js || 'off' === $cf_minify->css || 'off' === $cf_minify->html ) {
+				$cf_minify_value = 'off';
+			}
+
+			$cf_settings_array = [
+				'cache_level'       => $cache_level,
+				'minify'            => $cf_minify_value,
+				'rocket_loader'     => $rocket_loader,
+				'browser_cache_ttl' => $browser_cache_ttl,
+			];
+
+			return $cf_settings_array;
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'cloudflare_current_settings', $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Get Cloudflare IPs. No API validation needed, all exceptions returns the default CF IPs array.
+	 *
+	 * @since 2.8.21 Save IPs in a transient to prevent calling the API everytime
+	 * @since 2.8.16
+	 *
+	 * @author Remy Perona
+	 *
+	 * @return Object Result of API request if successful, default CF IPs otherwise
+	 */
+	public function get_cloudflare_ips() {
+		$cf_ips = get_transient( 'rocket_cloudflare_ips' );
+		if ( false !== $cf_ips ) {
+			return $cf_ips;
+		}
+
+		$cf_email   = $this->options->get( 'cloudflare_email', null );
+		$cf_api_key = defined( 'WP_ROCKET_CF_API_KEY' ) ? WP_ROCKET_CF_API_KEY : $this->options->get( 'cloudflare_api_key', null );
+
+		$this->cloudflare_facade->set_api_credentials( $cf_email, $cf_api_key, '', self::CF_USER_AGENT );
+		try {
+			$cf_ips = $this->cloudflare_facade->ips();
+
+			if ( empty( $cf_ips->success ) ) {
+				// Set default IPs from Cloudflare if call to Cloudflare /ips API does not contain a success.
+				// Prevents from making API calls on each page load.
+				$cf_ips = $this->get_default_ips();
+			}
+
+			set_transient( 'rocket_cloudflare_ips', $cf_ips, 2 * WEEK_IN_SECONDS );
+		} catch ( \Exception $e ) {
+			// Set default IPs from Cloudflare if call to Cloudflare /ips API fails.
+			// Prevents from making API calls on each page load.
+			$cf_ips = $this->get_default_ips();
+			set_transient( 'rocket_cloudflare_ips', $cf_ips, 2 * WEEK_IN_SECONDS );
+			return $cf_ips;
+		}
+
+		return $cf_ips;
+	}
+
+	/**
+	 * Get default Cloudflare IPs.
+	 *
+	 * @since  3.5
+	 * @author Soponar Cristina
+	 *
+	 * @return Object Default Cloudflare connecting IPs.
+	 */
+	public function get_default_ips() {
```

Comment:
```suggestion
	private function get_default_ips() {
```

It's only used internally, no need for it to be public